hazelcast/src/main/java/com/hazelcast/internal/nio/DataReader.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.nio;
+
+import com.hazelcast.internal.serialization.SerializationService;
+import com.hazelcast.nio.serialization.Data;
+
+import java.io.IOException;
+
+public interface DataReader {
```

Comment:
can we have a short javaDoc to tell reasoning behind introducing this interface?

Reply:
Added. Since this interface only makes sense as an extension of `DataInput` (already extended by `ObjectDataInput`), now `DataReader extends DataInput`.